src/java/org/apache/cassandra/index/sai/disk/v1/segment/SegmentBuilder.java

```diff
@@ -73,103 +67,53 @@ public abstract class SegmentBuilder
     private ByteBuffer minTerm;
     private ByteBuffer maxTerm;
 
-    final IndexTermType indexTermType;
+    final StorageAttachedIndex index;
     long totalBytesAllocated;
     int rowCount = 0;
     int maxSegmentRowId = -1;
 
-    public static class BlockBalancedTreeSegmentBuilder extends SegmentBuilder
+    public static class RamBufferSegmentBuilder extends SegmentBuilder
     {
-        private final byte[] scratch;
-        private final BlockBalancedTreeRamBuffer trieBuffer;
+        protected final SegmentRamBuffer segmentRamBuffer;
 
-        public BlockBalancedTreeSegmentBuilder(IndexTermType indexTermType, NamedMemoryLimiter limiter)
+        public RamBufferSegmentBuilder(StorageAttachedIndex index, NamedMemoryLimiter namedMemoryLimiter)
```

Comment:
nit: It's still `limiter` everywhere else...honestly I think that's enough to know what this is given the context. Maybe just use the shorter name for now?